### DISCUSSION TITLE
I just can't get it to work

### DISCUSSION THREAD

**Discussion by Lashandan** [Koenkk/zigbee2mqtt#21593] [General]
Perhaps my contribution is better placed here than in the Issues section

What happened?
I tried to install Zigbee2MQTT following the instructions of this colleague from Youtube (https://www.youtube.com/watch?v=4y_dDgo0i2g&t=203s) but it just doesn't work.
Mosquitto broker is installed.
User for mqtt is created.
ZHA is not configured.
In the configuration of Zigbee2Mqtt user and password are stored and the core-mosquitto:1883 etc.
The correct port of the SkyConnect stick is also specified, here I keep finding different statements.
But I have finally decided on port: /dev/ttyUSB0.
Testing whether I can write here was returned with success.
Countless restarts included.

I keep getting the error:

Zigbee2MQTT:info 2024-02-26 13:02:20: Starting Zigbee2MQTT version 1.35.3 (commit #unknown)
Zigbee2MQTT:info 2024-02-26 13:02:20: Starting zigbee-herdsman (0.33.8)
Zigbee2MQTT:error 2024-02-26 13:03:30: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2024-02-26 13:03:30: Failed to start zigbee
Zigbee2MQTT:error 2024-02-26 13:03:30: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2024-02-26 13:03:30: Exiting...
Zigbee2MQTT:error 2024-02-26 13:03:30: Error: Failure to connect
at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:364:19)
at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:158:13)
at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
at Zigbee.start (/app/lib/zigbee.ts:60:27)
at Controller.start (/app/lib/controller.ts:109:27)
at start (/app/index.js:107:5)

I urgently need help.
Can you provide me with all the information I need to solve my problem?
Many thanks in advance.

What did you expect to happen?
No response

How to reproduce it (minimal and precise)
No response

Zigbee2MQTT version
1.35.3-1

Adapter firmware version
dont know

Adapter
SkyConnect_v1.0

Setup
Addon

Debug log
[13:02:17] INFO: Preparing to start...
[13:02:17] INFO: Socat not enabled
[13:02:18] INFO: Starting Zigbee2MQTT...
Zigbee2MQTT:info 2024-02-26 13:02:19: Logging to console and directory: '/config/zigbee2mqtt/log/2024-02-26.13-02-19' filename: log.txt
Zigbee2MQTT:info 2024-02-26 13:02:20: Starting Zigbee2MQTT version 1.35.3 (commit #unknown)
Zigbee2MQTT:info 2024-02-26 13:02:20: Starting zigbee-herdsman (0.33.8)
Zigbee2MQTT:error 2024-02-26 13:03:30: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2024-02-26 13:03:30: Failed to start zigbee
Zigbee2MQTT:error 2024-02-26 13:03:30: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2024-02-26 13:03:30: Exiting...
Zigbee2MQTT:error 2024-02-26 13:03:30: Error: Failure to connect
at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:364:19)
at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:158:13)
at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
at Zigbee.start (/app/lib/zigbee.ts:60:27)
at Controller.start (/app/lib/controller.ts:109:27)
at start (/app/index.js:107:5)

**Comment by chris-1243** [comment]
According to your log, Z2M is not able to connect to your skyconnect dongle.

Could you provide a full debug log and your `configuration.yaml` ?

**Comment by Lashandan** [comment]
Hi Chris,

here is my /homeassistant/zigbee2mqtt/configuration.yaml

homeassistant: true
permit_join: false
mqtt:
 base_topic: zigbee2mqtt
 server: mqtt://core-mosquitto:1883
 user: mqtt_user
 password: mqtt_user
  
serial:
 port: /dev/serial/by-id/usb-Nabu_Casa_SkyConnect_v1.0_d4b486d1bb96ed11a2eec498a7669f5d-if00-port0
#adapter: ezsp

# Will run frontend on port 8080
frontend: true 

but what is still unclear is where I have to make the config. if I look directly into the addon Zigbee2Mqtt in the configuration, it looks like this. ![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/6a2fc2de-f2bc-4fc4-a621-4099e1c45f5c)
and here is my log:
/homeassistant/zigbee2mqtt/log/2024-02-28.08-41-08/log.txt

info  2024-02-28 08:41:08: Logging to console and directory: '/config/zigbee2mqtt/log/2024-02-28.08-41-08' filename: log.txt
info  2024-02-28 08:41:08: Starting Zigbee2MQTT version 1.35.3 (commit #unknown)
info  2024-02-28 08:41:08: Starting zigbee-herdsman (0.33.8)
error 2024-02-28 08:42:19: Error while starting zigbee-herdsman
error 2024-02-28 08:42:19: Failed to start zigbee
error 2024-02-28 08:42:19: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
error 2024-02-28 08:42:19: Exiting...
error 2024-02-28 08:42:19: Error: Failure to connect
    at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:364:19)
    at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:158:13)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:60:27)
    at Controller.start (/app/lib/controller.ts:109:27)
    at start (/app/index.js:107:5)


otherwise the procedure was as follows: I connected the SkyConnect stick. configured ZHA for the first time and then deactivated and uninstalled it again. then I did the configuration as described in many videos. without major code adjustments or terminal commands. I don't know what else I need to do, I've rummaged through all the documentation but haven't found anything to address my problem.

**Reply by chris-1243** [reply]
Could you uncomment this line in `configuration.yaml`

```
serial:
  adapter: ezsp
```

The driver needs to be specified

I do not used Z2M as an addon. I assumed, according your picture, `configuration.yaml` should go there /config/zigbee2mqtt

**Comment by Lashandan** [comment]
Okay, I have commented out the line with the adapter again.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/686597a5-b0b1-48ab-a6a8-48bb75b9ab3d)

Is it necessary to enter everything here again?

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/4b9609af-2101-44ea-816e-39f62e2383f5)

**Reply by chris-1243** [reply]
I would copy/paste all the details in your addon.

**Comment by Lashandan** [comment]
still the same result...

[09:23:06] INFO: Preparing to start...
[09:23:07] INFO: Socat not enabled
[09:23:07] INFO: Starting Zigbee2MQTT...
Zigbee2MQTT:info  2024-02-29 09:23:09: Logging to console and directory: '/config/zigbee2mqtt/log/2024-02-29.09-23-09' filename: log.txt
Zigbee2MQTT:info  2024-02-29 09:23:09: Starting Zigbee2MQTT version 1.35.3 (commit #unknown)
Zigbee2MQTT:info  2024-02-29 09:23:09: Starting zigbee-herdsman (0.33.8)
Zigbee2MQTT:error 2024-02-29 09:24:19: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2024-02-29 09:24:19: Failed to start zigbee
Zigbee2MQTT:error 2024-02-29 09:24:19: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2024-02-29 09:24:19: Exiting...
Zigbee2MQTT:error 2024-02-29 09:24:19: Error: Failure to connect
    at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:364:19)
    at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:158:13)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:60:27)
    at Controller.start (/app/lib/controller.ts:109:27)
    at start (/app/index.js:107:5)
    
    
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/796379a9-21b9-4d2a-9a86-e7694ff10d0c)

i have no idea how to fix this anymore...

I used this video as a template for the setup, but the whole config part is not discussed here either. Download here, install there, user password and everything works...
https://www.youtube.com/watch?v=4y_dDgo0i2g&t=296s

**Comment by chris-1243** [comment]
As you use special character in your mqtt password and according to the official documentation for the [addon](https://github.com/zigbee2mqtt/hassio-zigbee2mqtt?tab=readme-ov-file#installation), I would enclose the password by quotes.

It would be really helpful if you could provide the full debug log which is more detailed than those you provided.

> The correct port of the SkyConnect stick is also specified, here I keep finding different statements.
But I have finally decided on port: /dev/ttyUSB0

Why don't you let the system use the right port ? If your adapter is connected on an other port, like /dev/ttyACM0 why did you change it ?

**Comment by Lashandan** [comment]
Unfortunately I don't have a better log than what I posted, there is nothing more detailed under /zigbee2mqtt/

I tried the password and put everything in apostrophe:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/d0d83e98-d231-4517-975d-01740802029a)

The description from the Youtube video shows the path for the stick like this, but here I have also found all variations on the net as to which path you should specify - it worked for everyone else, but not one of them works for me ^^
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/8e4e55f0-8d93-4dbd-8696-6c8241cb0955)

When I just took the screenshot for the SkyStick, I noticed that under System-->Hardware it still says this
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/2268b1a0-30f0-442c-8d65-7de88192c3ee)

Do I still have to do this?
There was no mention of this anywhere in advance.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/bfd0ec34-322f-4c93-8479-c4d77de48740)

**Comment by chris-1243** [comment]
could you add this to your `configuration.yaml`

```
advanced:
  # Optional: Logging level, options: debug, info, warn, error (default: info)
  log_level: debug
``` 

Do not start the multi-protocol.

**Comment by Lashandan** [comment]
Thanks for your patience and for wanting to help.

I activated the log and found this:

[09:44:52] INFO: Preparing to start...
[09:44:53] INFO: Socat not enabled
[09:44:53] INFO: Starting Zigbee2MQTT...
Zigbee2MQTT:debug 2024-03-01 09:44:55: Loaded state from file /config/zigbee2mqtt/state.json
Zigbee2MQTT:info  2024-03-01 09:44:55: Logging to console and directory: '/config/zigbee2mqtt/log/2024-03-01.09-44-55' filename: log.txt
Zigbee2MQTT:debug 2024-03-01 09:44:55: Removing old log directory '/config/zigbee2mqtt/log/2024-03-01.09-32-44'
Zigbee2MQTT:info  2024-03-01 09:44:55: Starting Zigbee2MQTT version 1.35.3 (commit #unknown)
Zigbee2MQTT:info  2024-03-01 09:44:55: Starting zigbee-herdsman (0.33.8)
Zigbee2MQTT:debug 2024-03-01 09:44:55: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/config/zigbee2mqtt/coordinator_backup.json","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","databasePath":"/config/zigbee2mqtt/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"adapter":"ezsp","path":"/dev/ttyUSB0"}}'
Zigbee2MQTT:error 2024-03-01 09:46:05: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2024-03-01 09:46:05: Failed to start zigbee
Zigbee2MQTT:error 2024-03-01 09:46:05: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2024-03-01 09:46:05: Exiting...
Zigbee2MQTT:error 2024-03-01 09:46:05: Error: Failure to connect
    at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:364:19)
    at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:158:13)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:60:27)
    at Controller.start (/app/lib/controller.ts:109:27)
    at start (/app/index.js:107:5)


![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/3df708d4-b3a2-480d-bc21-5c4315d7583e)

**Reply by chris-1243** [reply]
Well, as your full log says and according to your screenshot, your adapter is definitely not connected.
```
xxx@rpi4:~ $ sudo dmesg | grep tty
[    0.000825] printk: console [tty1] enabled
[    1.549235] fe201000.serial: ttyAMA1 at MMIO 0xfe201000 (irq = 36, base_baud = 0) is a PL011 rev2
[    1.549418] serial serial0: tty port ttyAMA1 registered
[    5.292686] systemd[1]: Created slice system-getty.slice - Slice /system/getty.
[    8.389570] cdc_acm 1-1.3:1.0: ttyACM0: USB ACM device
xxx@rpi4:~ $ sudo dmesg | grep acm
[    8.389570] cdc_acm 1-1.3:1.0: ttyACM0: USB ACM device
[    8.390007] usbcore: registered new interface driver cdc_acm
[    8.390023] cdc_acm: USB Abstract Control Model driver for USB modems and ISDN adapters
``` 
Above, what I have connected on tty. For information, ttyACM0 is a Sonoff Dongle-E.

What is the output of this command `ls -l /dev/serial/by-id` ? The output should look like something like below:

```
xxx@rpi4:~ $ ls -la /dev/serial/by-id
total 0
drwxr-xr-x 2 root root 60 Mar  1 09:17 .
drwxr-xr-x 4 root root 80 Mar  1 09:17 ..
lrwxrwxrwx 1 root root 13 Mar  1 09:17 usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20230804184749-if00 -> ../../ttyACM0
```

**Comment by Lashandan** [comment]
I just don't understand it, I have connected the stick to the supplied extension.

This is what I get with your command:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/f64d61c0-493b-482b-954f-72b509426bb4)

**Reply by chris-1243** [reply]
ttyUSB0 is the right extension for sure and your dongle is recognized by your system which is good.

Your configuration looks pretty ok from my side. The last thing I see, an other program is connected to it (ZHA maybe ?) This is why Z2M is not able to connect to your adapter.

Could provide the output of `lsusb`

**Comment by Lashandan** [comment]
ZHA could be the problem if it was still there, I disabled it and finally uninstalled it. only then did I do all the setup for Zigbee2Mqtt. so ZHA is completely uninstalled I think ^^

lsusb:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/c0a01876-a231-4b8d-9a1a-500f4fa8a542)

**Reply by Lashandan** [reply]
ea60 should be my stick

**Comment by chris-1243** [comment]
At the beginning of your thread, you said you choose to go via ttyUSB. Did you make any commands to force such behavior ?

Definitely, your system does not want to recognized correctly your usb dongle. let's try `lsusb -v` to add some verbosity...

**Comment by Lashandan** [comment]
I don't really care what I use. 
These configurations were only disclosed to me in videos or instructions.

With lsusb -v I unfortunately don't get any more information. even with sudo lsusb -v I don't get any more...
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/03f48c92-435a-449c-9bc4-3268c7b761bf)


A look at the hardware shows me all this:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/7adddaaf-fcee-4265-acd7-7686f5d3a95f)

**Comment by chris-1243** [comment]
Let's try an other thing

`test -w /dev/ttyUSB0 && echo success || echo failure `

**Reply by Lashandan** [reply]
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/9cfc7116-0433-490f-9c81-1d7335df09a2)

**Comment by chris-1243** [comment]
I am starting to run out of idea.

In `configuration.yaml` let's try this

```
mqtt:
  server: mqtt://x.x.x.x:1883 # replace x.x.x.x by the ip adress of your home assistant computer
```

Your issue looks similar to one I saw already

**Comment by Lashandan** [comment]
Apologies for the delayed response. I changed the config to my IP, but no change; it just won't start. I'm slowly losing hope that this thing will work for me.

I connected the stick to my computer to ensure that it is recognized/works at all.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/edb4ee23-103e-4f78-bfd3-4ea88b45efec)

Seems to work...

But still, the device isn't working correctly with the addon. It's driving me crazy.

**Reply by chris-1243** [reply]
No worries. I used my spare time to run an install of haos on my test Pi to have a try. I wanted to understand how this addon works.

Here the configuration used and it is working. I had some trouble with the MQTT broker as I was not able to copy/paste an username/password.

```
data_path: /config/zigbee2mqtt
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: false
mqtt:
  server: mqtt://core-mosquitto:1883
  user: mqtt_test
  password: mqtt_passwd
serial:
  port: >-
    /dev/serial/by-id/usb-ITEAD_SONOFF_Zigbee_3.0_USB_Dongle_Plus_V2_20230804184749-if00
  adapter: ezsp
``` 
The output log:
```
[23:41:13] INFO: Preparing to start...
[23:41:14] INFO: Socat not enabled
[23:41:16] INFO: Starting Zigbee2MQTT...
Zigbee2MQTT:info  2024-03-02 23:41:21: Logging to console and directory: '/config/zigbee2mqtt/log/2024-03-02.23-41-21' filename: log.txt
Zigbee2MQTT:info  2024-03-02 23:41:21: Starting Zigbee2MQTT version 1.36.0 (commit #unknown)
Zigbee2MQTT:info  2024-03-02 23:41:21: Starting zigbee-herdsman (0.35.1)
Assertion failed: exportKey(EmberKeyType.CURRENT_NETWORK_KEY) returned unexpected SL status: 45
Zigbee2MQTT:info  2024-03-02 23:41:26: zigbee-herdsman started (resumed)
Zigbee2MQTT:info  2024-03-02 23:41:26: Coordinator firmware version: '{"meta":{"maintrel":"1 ","majorrel":"7","minorrel":"4","product":13,"revision":"7.4.1.0 build 0"},"type":"EZSP v13"}'
Zigbee2MQTT:info  2024-03-02 23:41:26: Currently 0 devices are joined:
Zigbee2MQTT:info  2024-03-02 23:41:26: Zigbee: disabling joining new devices.
Zigbee2MQTT:info  2024-03-02 23:41:26: Connecting to MQTT server at mqtt://core-mosquitto:1883
Zigbee2MQTT:info  2024-03-02 23:41:27: Connected to MQTT server
Zigbee2MQTT:info  2024-03-02 23:41:27: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload '{"state":"online"}'
Zigbee2MQTT:info  2024-03-02 23:41:27: Started frontend on port 8099
Zigbee2MQTT:info  2024-03-02 23:41:27: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload '{"state":"online"}'
Zigbee2MQTT:info  2024-03-02 23:41:27: Zigbee2MQTT started!
``` 

I changed the mqtt ip by the one of the Pi and it is working as well. Just a question, do you really use the username as a password in mqtt as you described in your first post ?

```
mqtt:
  user: mqtt_user
  password: mqtt_user
```

**Reply by Lashandan** [reply]
I changed the password now but yes, it was the password beforde.

i copied some parts of your config and now my config looks like this.


```
# Datapath
data_path: /config/zigbee2mqtt
# Frontend and HA
homeassistant: true
permit_join: false
# Will run frontend on port 8080
frontend: true
# zigbee_herdsman_debug
zigbee_herdsman_debug: true
# Socat
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: false
# Mqtt
mqtt:
 base_topic: zigbee2mqtt
 server: mqtt://192.168.178.36:1883
 user: mqtt_user
 password: 1234mqtt0901
# Serial
serial:
 port: /dev/ttyUSB0
 adapter: ezsp
# Advanced
advanced:
  # Optional: Logging level, options: debug, info, warn, error (default: info)
  log_level: debug

```

**Comment by Lashandan** [comment]
I changed the "herdsman debug-lvl"

i think there is the problem

```
2024-03-02T23:22:33.689Z zigbee-herdsman:adapter:ezsp:uart RTS/CTS config is off, enabling software flow control.
2024-03-02T23:22:33.689Z zigbee-herdsman:adapter:ezsp:uart Opening SerialPort with {"path":"/dev/ttyUSB0","baudRate":115200,"rtscts":false,"autoOpen":false,"parity":"none","stopBits":1,"xon":true,"xoff":true}
2024-03-02T23:22:33.695Z zigbee-herdsman:adapter:ezsp:uart Serialport opened
2024-03-02T23:22:33.695Z zigbee-herdsman:adapter:ezsp:uart Uart reseting
2024-03-02T23:22:33.695Z zigbee-herdsman:adapter:ezsp:uart --> Write reset
2024-03-02T23:22:33.696Z zigbee-herdsman:adapter:ezsp:uart --> [1ac038bc7e]
2024-03-02T23:22:33.697Z zigbee-herdsman:adapter:ezsp:uart -?- waiting reset
2024-03-02T23:22:43.703Z zigbee-herdsman:adapter:ezsp:uart --> Error: Error: {"sequence":-1} after 10000ms
2024-03-02T23:22:43.760Z zigbee-herdsman:adapter:ezsp:erro Connection attempt 1 error: Error: Failure to connect
    at SerialDriver.resetForReconnect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:350:19)
    at SerialDriver.emit (node:events:517:28)
    at /app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:348:22
    at Queue.execute (/app/node_modules/zigbee-herdsman/src/utils/queue.ts:35:20)
    at SerialDriver.openSerialPort (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:108:13)
    at SerialDriver.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:63:13)
    at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:356:17)
    at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:163:13)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:62:27)
2024-03-02T23:22:43.761Z zigbee-herdsman:adapter:ezsp:uart Port closed. Error? null
2024-03-02T23:22:43.761Z zigbee-herdsman:adapter:ezsp:ezsp onSerialClose()
2024-03-02T23:22:43.761Z zigbee-herdsman:adapter:ezsp:driv onEzspClose()
2024-03-02T23:22:43.761Z zigbee-herdsman:adapter:ezsp:debg onDriverClose()
2024-03-02T23:22:48.761Z zigbee-herdsman:adapter:ezsp:ezsp Next attempt 2
```

**Reply by Lashandan** [reply]
is there something with the baudrate?
"baudRate":115200

**Reply by chris-1243** [reply]
You could add `baudrate: 115200` there. According to your log, this is the baudrate used already.

It may be the hardware flow control... let's try to add this:

```
serial:
  baudrate: 115200
  rtscts: true
```

**Comment by Lashandan** [comment]
```
Zigbee2MQTT:debug 2024-03-03 00:38:38: Removing old log directory '/config/zigbee2mqtt/log/2024-03-03.00-19-18'
Zigbee2MQTT:info  2024-03-03 00:38:38: Starting Zigbee2MQTT version 1.36.0 (commit #unknown)
Zigbee2MQTT:info  2024-03-03 00:38:38: Starting zigbee-herdsman (0.35.1)
Zigbee2MQTT:debug 2024-03-03 00:38:38: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/config/zigbee2mqtt/coordinator_backup.json","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","databasePath":"/config/zigbee2mqtt/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"adapter":"ezsp","baudRate":115200,"path":"/dev/ttyUSB0","rtscts":true}}'
2024-03-02T23:38:38.706Z zigbee-herdsman:adapter Failed to validate path: 'Error: spawn udevadm ENOENT'
2024-03-02T23:38:38.706Z zigbee-herdsman:adapter:ezsp:debg Adapter concurrent: 8
2024-03-02T23:38:38.707Z zigbee-herdsman:controller:log Starting with options '{"network":{"networkKeyDistribute":false,"networkKey":[1,3,5,7,9,11,13,15,0,2,4,6,8,10,12,13],"panID":6754,"extendedPanID":[221,221,221,221,221,221,221,221],"channelList":[11]},"serialPort":{"baudRate":115200,"rtscts":true,"path":"/dev/ttyUSB0","adapter":"ezsp"},"databasePath":"/config/zigbee2mqtt/database.db","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","backupPath":"/config/zigbee2mqtt/coordinator_backup.json","adapter":{"disableLED":false,"concurrent":null,"delay":null}}'
2024-03-02T23:38:38.708Z zigbee-herdsman:adapter:ezsp:uart Opening SerialPort with {"path":"/dev/ttyUSB0","baudRate":115200,"rtscts":true,"autoOpen":false,"parity":"none","stopBits":1,"xon":false,"xoff":false}
2024-03-02T23:38:38.714Z zigbee-herdsman:adapter:ezsp:uart Serialport opened
2024-03-02T23:38:38.714Z zigbee-herdsman:adapter:ezsp:uart Uart reseting
2024-03-02T23:38:38.715Z zigbee-herdsman:adapter:ezsp:uart --> Write reset
2024-03-02T23:38:38.715Z zigbee-herdsman:adapter:ezsp:uart --> [1ac038bc7e]
2024-03-02T23:38:38.716Z zigbee-herdsman:adapter:ezsp:uart -?- waiting reset
2024-03-02T23:38:48.721Z zigbee-herdsman:adapter:ezsp:uart --> Error: Error: {"sequence":-1} after 10000ms
2024-03-02T23:38:48.798Z zigbee-herdsman:adapter:ezsp:erro Connection attempt 1 error: Error: Failure to connect
    at SerialDriver.resetForReconnect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:350:19)
    at SerialDriver.emit (node:events:517:28)
    at /app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:348:22
    at Queue.execute (/app/node_modules/zigbee-herdsman/src/utils/queue.ts:35:20)
    at SerialDriver.openSerialPort (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:108:13)
    at SerialDriver.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/uart.ts:63:13)
    at Ezsp.connect (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/ezsp.ts:356:17)
    at Driver.startup (/app/node_modules/zigbee-herdsman/src/adapter/ezsp/driver/driver.ts:163:13)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/app/lib/zigbee.ts:62:27)
2024-03-02T23:38:48.798Z zigbee-herdsman:adapter:ezsp:uart Port closed. Error? null
2024-03-02T23:38:48.798Z zigbee-herdsman:adapter:ezsp:ezsp onSerialClose()
2024-03-02T23:38:48.799Z zigbee-herdsman:adapter:ezsp:driv onEzspClose()
2024-03-02T23:38:48.799Z zigbee-herdsman:adapter:ezsp:debg onDriverClose()
``` 

:-(

**Reply by Lashandan** [reply]
I'm already toying with the idea of ​​simply configuring ZHA to test whether it works.

**Comment by Lashandan** [comment]
```
debug 2024-03-03 00:52:15: Loaded state from file /config/zigbee2mqtt/state.json
info  2024-03-03 00:52:15: Logging to console and directory: '/config/zigbee2mqtt/log/2024-03-03.00-52-15' filename: log.txt
debug 2024-03-03 00:52:15: Removing old log directory '/config/zigbee2mqtt/log/2024-03-03.00-48-43'
info  2024-03-03 00:52:15: Starting Zigbee2MQTT version 1.36.0 (commit #unknown)
info  2024-03-03 00:52:15: Starting zigbee-herdsman (0.35.1)
debug 2024-03-03 00:52:15: Using zigbee-herdsman with settings: '{"adapter":{"concurrent":null,"delay":null,"disableLED":false},"backupPath":"/config/zigbee2mqtt/coordinator_backup.json","databaseBackupPath":"/config/zigbee2mqtt/database.db.backup","databasePath":"/config/zigbee2mqtt/database.db","network":{"channelList":[11],"extendedPanID":[221,221,221,221,221,221,221,221],"networkKey":"HIDDEN","panID":6754},"serialPort":{"adapter":"ezsp","baudRate":115200,"path":"/dev/ttyUSB0","rtscts":true}}'
info  2024-03-03 00:52:20: zigbee-herdsman started (resumed)
info  2024-03-03 00:52:20: Coordinator firmware version: '{"meta":{"maintrel":"1 ","majorrel":"7","minorrel":"3","product":12,"revision":"7.3.1.0 build 176"},"type":"EZSP v12"}'
debug 2024-03-03 00:52:20: Zigbee network parameters: {"channel":11,"extendedPanID":221,"panID":6754}
info  2024-03-03 00:52:20: Currently 0 devices are joined:
info  2024-03-03 00:52:20: Zigbee: disabling joining new devices.
info  2024-03-03 00:52:20: Connecting to MQTT server at mqtt://192.168.178.36:1883
debug 2024-03-03 00:52:20: Using MQTT login with username: mqtt_user
error 2024-03-03 00:52:20: MQTT error: Connection refused: Not authorized
error 2024-03-03 00:52:20: MQTT failed to connect, exiting...
info  2024-03-03 00:52:20: Stopping zigbee-herdsman...
info  2024-03-03 00:52:21: Stopped zigbee-herdsman
```

**Reply by Lashandan** [reply]
For later research or people who come across the same error:

In my case, it was simply due to the wrong combination of user and password.

```
info  2024-03-03 00:52:20: Connecting to MQTT server at mqtt://192.168.178.36:1883
debug 2024-03-03 00:52:20: Using MQTT login with username: mqtt_user
error 2024-03-03 00:52:20: MQTT error: Connection refused: Not authorized
error 2024-03-03 00:52:20: MQTT failed to connect, exiting...
```

**Comment by Lashandan** [comment]
Ohhh my god - i finally changed the mqtt-user from: mqtt_user to mqtt-user and fu**ing finally it works:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20851108/ff999cba-ec23-4ed2-9a8d-8592d1f67dfa)

Thank you very much for your help and emotional support

**Reply by chris-1243** [reply]
Your log was showing it to you
`error 2024-03-03 00:52:20: MQTT error: Connection refused: Not authorized `

Human error😉. For my side, I'm glad to have provided some help. I learnt a lot as well and had a try to haos👍...

Now, what I suggest as it is working is to apply some tips in order to avoid futher problems.

- Check your wifi channel and adapt Zigbee in accordance. 

- Use an usb cable between your computer and your dongle (interferance).

- Avoid to connect your dongle to an ubs 3.0 port. If it is not possible, use an usb 2.0 cable well shielded or an usb hub.

https://www.zigbee2mqtt.io/advanced/zigbee/02_improve_network_range_and_stability.html

Take some times to read the website of Z2M. There are lots of advice which could help you a lot.

In `configuration.yaml`

```
serial:
  rtscts: true # it is optional
```

It should work without.

**Comment by chris-1243** [comment]
As it is working, may you closed your issue

**Comment by Lashandan** [comment]
Hi Chris,

Thanks again for your help.

I was also able to learn a lot from this human error for further HA projects.

In hindsight, I am also annoyed that it was due to the user account and at the same time I am glad that the error was so trivial.

Thank you also for the further tips.
I am already using the extension cable.
The cable in turn is plugged into the USB 2.0 port.

And finally I have set the "rtscts" value in the serial section back to false and it still works.

Thank you very much and that closes the case.
